### PR TITLE
log entire server response

### PR DESCRIPTION
**Pull request by makampf**

### CODE REVIEW

**Comment by makampf** [comment]
Linting error addressed in: https://github.com/github/super-linter/issues/2461

**Comment by makampf** [comment]
[LGMT](https://lgtm.com/help/lgtm/java-extraction) only supports Java version until 14 atm

**Comment by chgl** [comment]
> Linting error addressed in: [github/super-linter#2461](https://github.com/github/super-linter/issues/2461)

The illegal access seems to just be a warning, the actual fail is: https://github.com/miracum/kafka-fhir-to-server/runs/5141267263?check_suite_focus=true#step:5:223

> SendToServerProcessor.java:118:40: error: ')' expected

Might be an older version of Google java format unfamiliar with this new syntax.

We could go ahead and disable java linting as part of the super linter since we are already using spotless which will catch formatting issues when building. See https://github.com/miracum/fhir-gateway/blob/master/.github/workflows/ci.yml#L31 for disabling the linter.

Similarly, lgtm might just be disabled entirely until java 17 is supported (if ever). See https://github.com/organizations/miracum/settings/installations click 'configure' for the lgtm app and remove this repo from the ones the app should look at.

File: src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java [hunk 1]
```diff
@@ -95,7 +95,7 @@ public SendToServerProcessor(
           public <T, E extends Throwable> void onError(
               RetryContext context, RetryCallback<T, E> callback, Throwable throwable) {
             LOG.warn(
-                "Trying to sent resource to FHIR server caused error. {} attempt.",
+                "Trying to sent resource to FHIR server caused error. {} attempt. Error: {}",
```
Comment (chgl) [inline src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:98]: Throwable/exception as last Parameter already printed the exception including stacktrace, so it is preferable over formatting it as "{}". The problem is, that the exception message isn't particularly helpful. Instead, we should match on the exception type and if it's a fhirexception such as https://hapifhir.io/hapi-fhir/apidocs/hapi-fhir-base/ca/uhn/fhir/rest/server/exceptions/UnprocessableEntityException.html we should retrieve it's Operation outcome and either pretty print it as Json in it's entirety or manually list and log all issues returned by the server.

File: src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java [hunk 2]
```diff
@@ -115,10 +115,7 @@ public SendToServerProcessor(
       MDC.put("resourceId", resource.getIdElement().toUnqualifiedVersionless().toString());
       MDC.put("resourceType", resource.getResourceType().name());
 
-      Bundle bundle;
-      if (resource instanceof Bundle) {
-        bundle = (Bundle) resource;
-      } else {
+      if (!(resource instanceof Bundle bundle)) {
```
Comment (chgl) [inline src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:118]: Love it!

File: src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java [hunk 3]
```diff
@@ -135,8 +132,9 @@ public SendToServerProcessor(
             "Applying FHIR path filter {} to resource", kv("expression", fhirPathFilterExpression));
 
         shouldSend =
-            filterDurationTimer.record(
-                () -> resourceFilter.matches(resource, fhirPathFilterExpression));
+            Boolean.TRUE.equals(
```
Comment (chgl) [inline src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:135]: Not sure why this is needed?

File: src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java [hunk 4]
```diff
@@ -99,6 +100,14 @@ public SendToServerProcessor(
                 context.getRetryCount(),
                 throwable);
             sendingFailedCounter.increment();
+            if (throwable instanceof BaseServerResponseException fhirException) {
```
Comment (makampf) [inline src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:103]: So besser?

Comment (chgl) [reply src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:103]: Fehlt glaube noch ein Log von der Ausgabe in Zeile 105. *Falls* der Json blob der da zurückkommt schwer zu lesen ist in logausgaben kann man sonst die operationoutcomd.issues durchiterieren und einen loggen.

Comment (makampf) [reply src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:103]: 🙈 selbstverständlich fehlt der log.
Ich bin mir nicht so sicher, wie ich den am besten testen kann, sonst wäre es mir wahrscheinlich selbst aufgefallen.

Comment (chgl) [reply src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java:103]: Gute Frage. Ggf in unit test https://github.com/miracum/kafka-fhir-to-server/blob/master/src/main/java/org/miracum/streams/fhirtoserver/SendToServerProcessor.java#L65 mit gemocktem igenericclient der beim Ausführen der Transaktion eine exception wirft. Dann kann man zumindest manuell schnell sehen ob was geloggt wird - man kann theoretisch auch stdout capturen und asserten dass das was richtiges drinsteht... Lohnt sich aber vmtl nicht der Aufwand.

**Pull request merged**